modules/javafx.controls/src/main/java/javafx/scene/chart/Chart.java

```diff
@@ -540,40 +541,31 @@ final boolean isAccessibilityActive() {
             if (accessibilityActive instanceof SimpleBooleanProperty p) {
                 return p.get();
             } else {
-                boolean aa = Platform.accessibilityActiveProperty().get();
-                SimpleBooleanProperty active = new SimpleBooleanProperty(aa);
+                if (accessibilityActive instanceof Subscription sub) {
+                    sub.unsubscribe();
+                }
+                SimpleBooleanProperty active = new SimpleBooleanProperty();
                 accessibilityActive = active;
-                active.bind(Platform.accessibilityActiveProperty());
                 active.addListener((src, prev, on) -> {
                     handleAccessibilityActive(on);
                 });
+                active.bind(Platform.accessibilityActiveProperty());
                 return active.get();
             }
         } else {
             // chart and its data are allowed to be constructed in a background thread
             if (accessibilityActive == null) {
-                // set up a listener which, once the chart gets connected to a window (always in the context of
-                // the fx application thread), calls isAccessibilityActive() again,
-                // which in turn installs the listener on the Platform.accessibilityActiveProperty.
-                ObservableValue<Window> winProp = sceneProperty().flatMap(Scene::windowProperty);
-                accessibilityActive = winProp; // keep the reference so it won't get gc
-
-                // lambda cannot be used in place of a ChangeListener in removeListener()
-                ChangeListener<Window> li = new ChangeListener<>() {
-                    @Override
-                    public void changed(ObservableValue<? extends Window> src, Window old, Window win) {
-                        if (win != null) {
-                            if (accessibilityActive == winProp) {
-                                accessibilityActive = null;
-                            }
+                // set up a subscription to be fired once the chart becomes a part of the scene graph
+                accessibilityActive = sceneProperty()
+                    .flatMap(Scene::windowProperty)
+                    .subscribe((w) -> {
+                        if (w != null) {
+                            // also unsubscribes when appears in a window
```

Comment:
Suggestion: Add a comment that this can only happen on the FX app thread, since you are relying on that.

Reply:
added comment